### ISSUE TITLE
LwM2mSingleResource constructor can create invalid resources

### ISSUE THREAD

**Issue by mhuelfen** [eclipse/leshan#182]
The LwM2mSingleResource constructor `newResource(int id, Object value, Type type)` can produce resources with type not matching its value.

Because the type points to the start point in the switch clause, so if the value matches a later condition a resource of this type is created.

e.g.
`newResource(0, "string",BOOLEAN)` creates a resource of type BOOLEAN with a string value

https://github.com/eclipse/leshan/blob/58c87fc50630217be07505b514525202b5c168f9/leshan-core/src/main/java/org/eclipse/leshan/core/node/LwM2mSingleResource.java#L44
``` 
    public static LwM2mSingleResource newResource(int id, Object value, Type type) {
        switch (type) {
        case INTEGER:
            if (value instanceof Long)
                break;
        case FLOAT:
            if (value instanceof Double)
                break;
        case BOOLEAN:
            if (value instanceof Boolean)
                break;
        case OPAQUE:
            if (value instanceof byte[])
                break;
        case STRING:
            if (value instanceof String)
                break;
        case TIME:
            if (value instanceof Date)
                break;
            throw new IllegalArgumentException("Value does not match the given datatype");
        default:
            throw new IllegalArgumentException(String.format("Type %s is not supported", type.name()));
        }
        return new LwM2mSingleResource(id, value, type);
    }
```

**Comment by sbernard31** [comment]
It should be fixed on master now (see commit f1982ed57468aaa86406972a1bec2ca9c15a8342)
Thx for reporting this.

**Comment by mhuelfen** [comment]
Thx looks good.
